examples/gno.land/r/demo/boards2/board.gno

```diff
@@ -96,21 +104,38 @@ func (board *Board) DeleteThread(pid PostID) {
 	}
 }
 
-func (board *Board) Render() string {
-	s := "\\[" + newLink("post", board.GetPostFormURL()) + "]\n\n"
-	if board.threads.Size() > 0 {
-		board.threads.Iterate("", "", func(_ string, v interface{}) bool {
-			post := v.(*Post)
-			if post.isHidden {
-				return false
-			}
+// Render renders a board into Markdown.
+//
+// Pager is used for pagination if it's not nil.
+func (board *Board) Render(p *PaginationOpts) string {
```

Comment:
We could anticipate different page sizes or offsets in each context and can introduce separate pagination objects (boards vs. thread, etc.) or have separate functions (long term).

Reply:
@salmad3 

> We could anticipate different page sizes or offsets in each context

Already, see `render.gno` file. There is different page size per category.